cli_tools/gce_vm_image_import/importer/processor.go

```diff
@@ -37,24 +37,15 @@ type processorProvider interface {
 
 type defaultProcessorProvider struct {
 	ImportArguments
-	imageClient   createImageClient
+	computeClient daisyCompute.Client
 	diskInspector disk.Inspector
 }
 
 func (d defaultProcessorProvider) provide(pd persistentDisk) (processor, error) {
 	if d.DataDisk {
-		return newDataDiskProcessor(pd, d.imageClient, d.Project,
+		return newDataDiskProcessor(pd, d.computeClient, d.Project,
 			d.Labels, d.StorageLocation, d.Description,
 			d.Family, d.ImageName), nil
 	}
-	if d.Inspect && d.diskInspector != nil {
-		log.Printf("Running experimental disk inspections on %v.", pd.uri)
-		inspectionResult, err := d.diskInspector.Inspect(pd.uri)
-		if err != nil {
-			log.Printf("Inspection error=%v", err)
-		} else {
-			log.Printf("Inspection result=%v", inspectionResult)
-		}
-	}
-	return newBootableDiskProcessor(d.ImportArguments, pd)
+	return newBootableDiskProcessor(d.computeClient, d.diskInspector, d.ImportArguments, pd)
```

Comment:
My original vision was a modular approach where inspection results determine which processor to return. For example, if the image is inspected as non-UEFI and Linux, then the additional code from this PR isn't required.

Benefits:
-  Individual processors are smaller, and easier to grok / test
-  Future development won't continue to increase the size of bootable disk processor

Tactically, to make that happen, a couple of options:
1. This provider returns a list of providers. The consumer iterates over each, and passes the processed disk to the subsequent step.
2. Don't change the signature here; instead, use a facade that iterates over each sub-processor.

Psuedocode

```

if data_disk:
 return  [data_disk_processor]

run inspection

processors := []

if data disk and windows:
  processors += uefi_tagger

processors += bootable_disk_processor
return processors
```

Reply:
See my new imp. After evaluation, a separate inspector makes the bst sense to me.

Reply:
Nice -- I think it's heading in the right direction. The current implementation that you have, though, mixes two *distinct* responsibilities: inspection and mutation. The tactical reason for separating these, is that we need the inspection results to make other decisions (which translator to run, for example), and therefore need inspection to finish prior to running your mutation steps.